src/plugin.tsx

```diff
@@ -274,16 +315,20 @@ export class TranscriptPlugin implements OnMediaUnload, OnMediaLoad, OnPluginSet
     };
 
     private _handleDownload = () => {
-        if (!this._captionsRaw) {
-            return;
+        const { playerConfig } = this._configs;
+        if (this._captions) {
+            const entryMetadata = playerConfig.sources.metadata;
+            downloadContent(
+                makePlainText(this._captions),
+                `${this._transcriptLanguage}${entryMetadata ? `-${entryMetadata.name}` : ""}.txt`
```

Comment:
this is checking if we have a metadata object and if so it uses the .name. It assumes that the metadata has a name (which is usually OK but sometimes there are entries without name). 
If we want to protect the logic we should check the metadata.name